### ISSUE TITLE
Can I add a custom overlay?

### ISSUE THREAD

**Issue by pgwzzz** [react-community/react-native-maps#1030]
Can I add a custom overlay on the map like the following?

https://developers.google.com/maps/documentation/javascript/examples/overlay-simple

I hope I can. 

Thank you very much.

**Comment by markmssd** [comment]
I've been asking the same question, see #868 ...

**Comment by pgwzzz** [comment]
@markmssd 

Hello.

I really need custom overlay function in react-native-maps but I think it's not implemented yet or it's hard to be implemented with some reason.

I decided to use Javascript Google Maps API in <WebView> like the link I attached although I think it's not best practice.

The result is nice enough. You should try it also.

**Comment by markmssd** [comment]
Hey @pgwzzz ,

Have you tried using a library such as [google-map-react](https://github.com/istarkov/google-map-react) within a WebView, e.g.
```<WebView source={here}></WebView>```?

If not, how was your approach?

**Comment by proProbe** [comment]
Im also looking to have somekind of a custom overlay! doesnt seem to be any progress on that part though. any ideas how to do it native?

**Comment by chocky335** [comment]
Hi there, @markmssd @proProbe @pgwzzz !
I started resolving this issue, firstly with ios default map, now you can look to my fork (https://github.com/chocky335/react-native-maps) and check it, maybe you'll help me with some issues(displaying mirrored images, to see image you need swipe or scale map).

**Comment by markmssd** [comment]
@chocky335 That's awesome!

As a temporary solution, I'm simply using a WebView with Google Maps into it. I'm very tight on time until april 6th, after which I will definitely give you a hand to finish it, and open a PR to have your changes merged into react-native-maps!

**Comment by dbrugne** [comment]
You can probably achieve that with MapView.UrlTile : [](https://github.com/airbnb/react-native-maps#using-a-custom-tile-overlay)

**Comment by markmssd** [comment]
This is what I ended up doing. It's also much faster, because my overlays are 7-8MB SVG files, which is horrible for performance...

**Comment by pjamrozowicz** [comment]
@markmssd @dbrugne Do you happen to know how to create heatmap (to display pollution on the map) using react-native-maps which would work both for Android and iOS? I know there is cutom-tiles option but I don't have any heatmap tiles generator.

**Comment by fmelp** [comment]
+1

**Comment by markmssd** [comment]
Sorry, I've never used heatmap... What you could do is use [MapTiler](www.maptiler.com) to generate your tiles, and host them on Amazon S3/CloudFront. Then, use the custom-tiles, pointing to your CloudFront endpoint. Not the best solution, but it worked amazingly for me to throw out a prototype quickly.

**Comment by fmelp** [comment]
@markmssd hi Mark, do you have any advice on how to overlay on image without using MapTiler. I have a hacky implementation for it by using Markers, but the resizing by zoom amount only works on iOS. Any chance that image overlay will be included in the pkg? Thanks!!

**Comment by shaikhussian** [comment]
+1

**Comment by Stophface** [comment]
@markmssd why would you say its not the best solution?!

**Comment by markmssd** [comment]
@Stophface simply because it's serving through CDN, instead of a server. However, I've tried hosting the tiles on MapBox Tilesets, and the performance was much slower... so it wasn't that much of a bad solution after all hehe. I also wanted to try it on Google's CDN, but haven't had time to do so yet.

**Comment by Stophface** [comment]
I am using `<MapView.UrlTile .../> to add a `Custom Tile Overlay`.

```
<MapView 
   ...>
  <MapView.UrlTile
    urlTemplate={this.state.urlTemplate}
  />
</MapView>
```

 It sort of works. There are two problems:

1. When I pinch, drag and zoom in/out, it displays the underlaying tiles (on Android Google Maps). Once the pinching, dragging, zooming is done, it only then loads the `Tiles` passed with `<MapView.UrlTile .../>`. It looks like basically it loads two maps. First the underlaying, and then the one passed through `<MapView.UrlTile .../>`.
Here a Video: [https://www.file-upload.net/download-12828114/maps_prob1.mov.html](https://www.file-upload.net/download-12828114/maps_prob1.mov.html) (19.41 MB, download link)

2. The names of the underlaying map shine through. Looking at the image I can see "Nashville", "Louisville", .... All the names shining through. Why is that?
[https://ibb.co/jXP6q6](https://ibb.co/jXP6q6)

**Comment by Stophface** [comment]
@markmssd any ideas regarding this https://github.com/airbnb/react-native-maps/issues/103 ?

**Comment by Stophface** [comment]
@markmssd any ideas regarding this: https://github.com/airbnb/react-native-maps/issues/1555

**Comment by alvelig** [comment]
Duplicate of #103

**Comment by coxato** [comment]
any updates? like an easier way to do it?
